### PR TITLE
Check if added location is within graph bounds.

### LINKED ISSUE #108: Check if location is within graph bounds before adding it

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Closes #108.
Instead of only geocoding within a bounding box, check geocoded co-ordinates against
OTP graph bounds polygon.  This way, all addresses returned by search service can be geocoded,
and only locations that can be routed to will be added.

### CODE REVIEW

File: src/nih_wayfinding/app/scripts/views/locations/add-location/location-profile/location-profile-controller.js [hunk 1]
```diff
@@ -82,13 +82,26 @@
          */
         function onGeocoderResponse(data) {
             if (data.length) { // If non-empty result
-                // Add geometric features to location tracker
-                ctl.user.setTempLocationProperty('feature', data[0]);
+                // check if location is within graph bounds
+                MapControl.getGraphBounds().then(function(geojson) {
```
Comment (CloudNiner) [inline src/nih_wayfinding/app/scripts/views/locations/add-location/location-profile/location-profile-controller.js]: It probably makes more sense to perform this check in the geocoder find response, and reject the promise with an error message there. That way, we get this bounds check both here and in the typeahead on the locations view.

**Comment by flibbertigibbet** [comment]
Okay, I think it's sorted now.  The geocoder does both the bounds and empty-response checks now, and will reject with a message for either.

File: src/nih_wayfinding/app/scripts/geocoder/search-service.js [hunk 2]
```diff
@@ -60,10 +59,26 @@
...
+                        var geom = response[0].geometry;
+                        // must use turf's geojson-y objects for turf.inside
+                        var point = turf.point([geom.x, geom.y]);
+                        var polygon = turf.polygon(geojson.coordinates);
+                        if (turf.inside(point, polygon)) {
+                            dfd.resolve(response);
+                        } else {
+                            dfd.reject('Address is outside the routing bounds. Please choose a different address.');
+                        }
+                    });
```
Comment (CloudNiner) [inline src/nih_wayfinding/app/scripts/geocoder/search-service.js:81]: Sorry I missed this earlier, but we should probably handle the case here where getGraphBounds fails, and just return the response without doing the graph bounds check.

**Comment by CloudNiner** [comment]
+1 assuming tests pass locally

Probably worth handling the getGraphBounds failure case

**Comment by flibbertigibbet** [comment]
Tests passed on Travis for the last push (and locally), so I'm not sure why GitHub is confused about that.

I'll add a check for graph bounds failure (would likely only happen if OTP wasn't running).

**Pull request merged**